base/util/src/main/java/com/netscape/cmsutil/crypto/CryptoUtil.java

```diff
@@ -571,6 +581,37 @@ public static KeyPair generateRSAKeyPair(CryptoToken token, int keysize, boolean
         return kg.genKeyPair();
     }
 
+    public static KeyPair generateRSAKeyPair(CryptoToken token, int keysize,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage usages[],
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage usages_mask[]) throws Exception {
+        return generateRSAKeyPair(token, keysize, false,false,false,usages,usages_mask);
+    }
+
+    public static KeyPair generateRSAKeyPair(CryptoToken token, int keysize, boolean temporary,
+            boolean sensitive, boolean extractable,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage usages[],
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage usages_mask[]) throws Exception {
+        KeyPairGenerator kg = token.getKeyPairGenerator(KeyPairAlgorithm.RSA);
+
+        if(usages != null) {
+            kg.setKeyPairUsages(usages, usages_mask);
+        }
+
+        if(extractable == true)
```

Comment:
Could simplify all this to `kg.extractablePairs(extractable)` &c -- moves the default into CryptoUtil versus JSS/NSS.